offer-manager-domain/src/main/groovy/life/qbic/business/offers/Offer.groovy

```diff
@@ -261,28 +261,35 @@ class Offer {
      * @return The net offer price including overheads
      */
     double getTotalNetPriceWithOverheads() {
-        return calculateNetPrice() - getTotalDiscountAmount() + getOverheadSum()
+        return (calculateNetPrice() + getOverheadSum()) as BigDecimal
     }
 
     /**
      * The overhead price amount of all service items without VAT.
      *
      * @return The calculated overhead amount of the selected items.
      */
-    double getOverheadSum() {
+    BigDecimal getOverheadSum() {
         if (items.empty) {
             return 0
         }
-        return items.sum {calculateItemOverhead(it)} as double
+        return items.sum {calculateItemOverhead(it)} as BigDecimal
     }
 
-    private BigDecimal calculateItemNet(ProductItem item) {
+    private BigDecimal determineUnitPrice(ProductItem item) {
```

Comment:
nice :+1: this might clash with the function that @jenniferboedker introduced in her PR.